add-chain/config/config.go

```diff
@@ -15,31 +15,6 @@ import (
 	"github.com/ethereum-optimism/superchain-registry/superchain"
 )
 
-type JSONChainConfig struct {
-	ChainID                          uint64                   `json:"l2_chain_id"`
-	BatchInboxAddr                   superchain.Address       `json:"batch_inbox_address"`
-	Genesis                          superchain.ChainGenesis  `json:"genesis"`
-	BlockTime                        uint64                   `json:"block_time"`
-	SequencerWindowSize              uint64                   `json:"seq_window_size"`
-	PlasmaConfig                     *superchain.PlasmaConfig `json:"plasma_config,omitempty"`
-	superchain.HardForkConfiguration `json:",inline"`
-}
-
-func (c *JSONChainConfig) VerifyPlasma() error {
-	if c.PlasmaConfig != nil {
-		if c.PlasmaConfig.DAChallengeAddress == nil {
-			return fmt.Errorf("missing required field: da_challenge_contract_address")
-		}
-		if c.PlasmaConfig.DAChallengeWindow == nil {
-			return fmt.Errorf("missing required field: da_challenge_window")
-		}
-		if c.PlasmaConfig.DAResolveWindow == nil {
-			return fmt.Errorf("missing required field: da_resolve_window")
-		}
-	}
-	return nil
-}
-
```

Comment:
Was able to get rid of this extraneous `JSONChainConfig` struct by just transferring the same `json` tags to the `superchain.ChainConfig` struct. Also moved the `JSONChainConfig.VerifyPlasma()` method to `superchain.ChainConfig.CheckDataAvailability()`